packages/core/src/entrypoint/index.ts

```diff
@@ -0,0 +1,190 @@
+>(
+  entryPoint: EntryPointRegistry[EntryPointVersion],
+  userOperation: UserOperationLike,
+  throwable?: boolean
+): UserOperationLike<TEntryPointVersion> | undefined => {
+  if (entryPoint.isUserOpVersion(userOperation)) {
+    return userOperation as UserOperationLike<TEntryPointVersion>;
+  }
+  if (!throwable) {
+    return undefined;
+  }
+  throw new MismatchingEntryPointError(entryPoint.version, userOperation);
+};
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends DefaultEntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options?: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<TChain extends Chain = Chain>(
+  chain: TChain,
+  options?: GetEntryPointOptions<DefaultEntryPointVersion>
+): EntryPointDefRegistry<TChain>[DefaultEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[EntryPointVersion] {
+  const { version = defaultEntryPointVersion, addressOverride } = options ?? {
+    version: defaultEntryPointVersion,
+  };
+
+  const entryPoint = entryPointRegistry[version ?? defaultEntryPointVersion];
+  const address = addressOverride || entryPoint.address[chain.id];
+  if (!address) {
+    throw new EntryPointNotFoundError(chain, version);
+  }
+
+  if (version === "0.6.0") {
+    const v6 = entryPoint as SupportedEntryPoint<
+      "0.6.0",
+      TChain,
+      typeof EntryPointAbi_v6
+    >;
+    return {
+      version: v6.version,
+      address,
+      chain,
+      abi: v6.abi,
+      getUserOperationHash: (r) =>
+        v6.getUserOperationHash(r, address, chain.id),
+      packUserOperation: v6.packUserOperation,
+      isUserOpVersion: v6.isUserOpVersion,
+      coerce: (userOperation, throwable) =>
+        coerceEntryPointVersion(v6, userOperation, throwable),
+    } as EntryPointDef<"0.6.0", TChain, typeof EntryPointAbi_v6>;
+  } else if (version === "0.7.0") {
+    const v7 = entryPoint as SupportedEntryPoint<
+      "0.7.0",
+      TChain,
+      typeof EntryPointAbi_v7
+    >;
+    return {
+      version: v7.version,
+      address,
+      chain,
+      abi: v7.abi,
+      getUserOperationHash: (r) =>
+        v7.getUserOperationHash(r, address, chain.id),
+      packUserOperation: v7.packUserOperation,
+      isUserOpVersion: v7.isUserOpVersion,
+      coerce: (userOperation, throwable) =>
+        coerceEntryPointVersion(v7, userOperation, throwable),
+    } as EntryPointDef<"0.7.0", TChain, typeof EntryPointAbi_v7>;
```

Comment:
same here should use `satisfies` instead of `as`

Reply:
no you need to use as. satisfies does work here.

Reply:
if satisfies isn't working it's because the type doesn't match the expected output... we should fix that so it does work

Reply:
see https://github.com/alchemyplatform/aa-sdk/pull/549